src/chart/generateCategoricalChart.tsx

```diff
@@ -859,11 +884,31 @@ export const generateCategoricalChart = ({
       const numericAxisId = item.props[`${numericAxisName}Id`];
       // axisId of the categorical axis
       const cateAxisId = item.props[`${cateAxisName}Id`];
-      const axisObj = axisComponents.reduce((result: any, entry: BaseAxisProps) => {
+
+      type AxisObj = {
+        xAxis?: BaseAxisProps;
+        xAxisTicks?: Array<TickItem>;
+
+        yAxis?: BaseAxisProps;
+        yAxisTicks?: Array<TickItem>;
+
+        zAxis?: BaseAxisProps;
+        zAxisTicks?: Array<TickItem>;
+
+        angleAxis?: BaseAxisProps;
+        angleAxisTicks?: Array<TickItem>;
+
+        radiusAxis?: BaseAxisProps;
+        radiusAxisTicks?: Array<TickItem>;
+      };
```

Comment:
Any reason to define this type inside of the function? Not against it, just wondering. 
If this object is something that is passed around, we might reuse it elsewhere. 🤔

Reply:
yeah +1, generally types aren't defined in functions even if they aren't reused

Reply:
🤷 I was just going for the smallest scope required. I can move it.

Reply:
Moved